### PR TITLE
Refactoring pomodoro-timer.js 

### PR DESCRIPTION
**Pull request by milosevicj**
- pomodoro-timer.js refactored
- added new more accurate timer tock
- configured maven to automatically run bower through npm (locally)

### CODE REVIEW

**Comment by stanislavgrujic** [comment]
@asuletic Could you please do the review?

File: src/main/webapp/resources/core/js/pomodoro-timer.js [hunk 1]
```diff
@@ -1,162 +1,121 @@
...
+    $('#shortButton').on('click', PeriodEnum.SHORT_BREAK.initialize);
+    $('#longButton').on('click', PeriodEnum.LONG_BREAK.initialize);
+    
+    $('#resetButton').on('click', resetTimer);
+    $('#playButton').on('click', playTimer);
+    $('#pauseButton').on('click', pauseTimer);
+    
+    function createPeriod(btnId){
+      var buttonId = btnId;
+      var self = this;
```
Comment (PavleP) [inline src/main/webapp/resources/core/js/pomodoro-timer.js]: Where variable self has been used?

File: src/main/webapp/resources/core/js/pomodoro-timer.js [hunk 2]
```diff
@@ -1,162 +1,121 @@
...
+      $('#minutesValue').text(minutes);
+      $('#secondsValue').text(seconds);
+    }
+    
+    function formatTime(time) {
+      return time > 9 ? time.toString() : '0' + time.toString();
+    }
+
   });
 })($);
```
Comment (PavleP) [inline src/main/webapp/resources/core/js/pomodoro-timer.js]: Please check file formatting. Fore example, on more places there is "){", but should be ") {".

File: src/main/webapp/resources/core/js/pomodoro-timer.js [hunk 3]
```diff
@@ -1,162 +1,121 @@
...
+    $('#pomodoroButton').on('click', PeriodEnum.POMODORO.initialize);
+    $('#shortButton').on('click', PeriodEnum.SHORT_BREAK.initialize);
+    $('#longButton').on('click', PeriodEnum.LONG_BREAK.initialize);
+    
+    $('#resetButton').on('click', resetTimer);
+    $('#playButton').on('click', playTimer);
+    $('#pauseButton').on('click', pauseTimer);
+    
+    function createPeriod(btnId){
+      var buttonId = btnId;
```
Comment (asuletic) [inline src/main/webapp/resources/core/js/pomodoro-timer.js]: Why you created new variable? You could use btnId.

File: src/main/webapp/resources/core/js/pomodoro-timer.js [hunk 4]
```diff
@@ -1,162 +1,121 @@
...
+      var self = this;
+      
+      var _initializeTimer = function(bulkMinutes){
+        gTimer.stop();
+        gTimer.start(bulkMinutes * 60 * 1000 + 10);
+        gTimer.pause();
+      }
+
```
Comment (asuletic) [inline src/main/webapp/resources/core/js/pomodoro-timer.js]: Button ids  (#pomodoroButton, #shortButton, #pauseButton,...) are used many times. Create constants for them. Also you can grab information about id from click event. Maybe you could use it

File: src/main/webapp/resources/core/js/pomodoro-timer.js [hunk 5]
```diff
@@ -1,162 +1,121 @@
...
-var audio = new Audio('/resources/core/sounds/beep.mp3');
-
 (function($) {
   $(function() {
-    pomodoroDuration = $('#pomodoroButton').attr('data-value');
-    time = bulkMinutesToTime(pomodoroDuration);
+    var gAudio = new Audio('/resources/core/sounds/beep.mp3');
+    var gPeriod;
+    var PeriodEnum = {
+        POMODORO: createPeriod('#pomodoroButton'),
```
Comment (asuletic) [inline src/main/webapp/resources/core/js/pomodoro-timer.js]: You assign function to properties POMODORO, SHORT_BREAK, LONG_BREAK. This is common for constants not for functions. Should be lower case.

File: src/main/webapp/resources/core/js/pomodoro-timer.js [hunk 6]
```diff
@@ -1,162 +1,121 @@
...
+      toggleActionButtons('#playButton', '#resetButton');
+      gTimer.pause();
+    }
+    
+    function toggleActionButtons() {
+      $('#resetButton').prop('disabled', true);
+      $('#playButton').prop('disabled', true);
+      $('#pauseButton').prop('disabled', true);
+     
+      $.each(arguments, function(index, buttonId) {
```
Comment (PavleP) [inline src/main/webapp/resources/core/js/pomodoro-timer.js]: > The arguments object is an Array-like object corresponding to the arguments passed to a function.

[link](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/arguments)

Comment (asuletic) [inline src/main/webapp/resources/core/js/pomodoro-timer.js]: I thought it should have at least args param:
function toggleActionButtons(...args){...}, but JS "swallows" that without args :-)
My mistake.

**Pull request merged**